Release v2.4.0

peterk87 opened the pull request:
Change log for v2.4.0 release:

- added scheme metadata table to Python package data (#109)
- typing scheme name change from `tb_speciation` to `tb_lineage` (#110)
- updated README (#111)
- updated default low coverage depth value for SE scheme from 50 to 20, max kmer frequency from 1000 to 10,000 and updated help/usage docs (#114)
- improved reverse complement function performance (#118)
- fixed merging of subtyping results with subtype result metadata table (#120)
- added CI testing for Python 3.7, cleaned up setup.py (#119)

DarianHole commented:
Hi @peterk87,

Everything seems to be in order on the code side. Pytests run on both 3.6 and 3.7 as expected.

Although I think I've found an installation issue I missed last time (or I don't know how to install). Currently I pip install from the branch (`pip install .`) and when I do that, the aho_corasick and qc directories are not being copied with the data (no issue when I tried 2.3 or 2.2 in this).

I'll see if its on my end a bit more as well

peterk87 commented on bio_hansel/main.py:
```diff
@@ -52,12 +46,12 @@ def init_parser():
                         help='Custom user-specified SNP substyping scheme name')
     parser.add_argument('-M', '--scheme-metadata',
-                        help='Scheme subtype metadata table (CSV or tab-delimited format; must contain "subtype" column)')
+                        help='Scheme subtype metadata table (.TSV format accepted; contain column called "subtype")')
```
Made it more explicit in f653c2f6aef0c8a93b2553de9bf926940894bafe with **MUST** and a bit more explanation about accepted file formats

peterk87 commented on README.rst:
```diff
@@ -255,7 +264,18 @@ Analysis of all FASTA/FASTQ files in a directory
+
+    hansel -s heidelberg -M <metadata.tsv> -vv --threads <n_cpu> -o results.tab -O match_results.tab -D /path/to/fastas_or_fastqs/
+
+``biohansel`` works best on TSV metadata files. If possible, use a tab separated metadata file or your analysis may fail. 
```
I've added more documentation here in f653c2f6aef0c8a93b2553de9bf926940894bafe with tables of accepted file formats and a very small example table.

peterk87 commented on setup.py:
```diff
@@ -3,50 +3,47 @@
+    name='bio_hansel',
+    package_data={'bio_hansel': ['data/*/*.fasta', 'data/*/*.tsv',]},
+    package_dir={'bio_hansel': 'bio_hansel'},
+    packages=find_packages(include=['bio_hansel']),
```
Thanks for catching this! Fixed in f653c2f6aef0c8a93b2553de9bf926940894bafe

peterk87 commented:
Thanks a lot @DarianHole for taking a look at this PR! I fixed the packaging issue and made the suggested changes. 

I think it *should* auto-deploy when a release tag is created. 

https://github.com/phac-nml/biohansel/blob/f653c2f6aef0c8a93b2553de9bf926940894bafe/.travis.yml#L13-L23

The user and secure password could be changed to the `phac-nml` and a secure password generated by the travis-ci cli tool. I've added the `phac-nml` user as a project owner on PyPI.

DarianHole commented:
@peterk87 I am not authorized to merge this one but its good. Let me know if anything else is needed!

peterk87 commented:
Thanks a lot @DarianHole! I'll create a v2.4.0 release and make sure it's up on PyPI

glabbe commented:
Thanks a lot @peterk87 and @DarianHole for this new version release, much appreciated!
